sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -661,7 +661,8 @@ <T> CosmosPagedFlux<T> readAllItems(CosmosQueryRequestOptions options, Class<T>
                 client,
                 queryOptionsAccessor.getQueryNameOrDefault(requestOptions, this.readAllItemsSpanName),
                 requestOptions.getConsistencyLevel(),
-                client.getEffectiveDiagnosticsThresholds(queryOptionsAccessor.getDiagnosticsThresholds(requestOptions)));
+                client.getEffectiveDiagnosticsThresholds(queryOptionsAccessor.getDiagnosticsThresholds(requestOptions)),
+                queryOptionsAccessor.getDiagnosticsFactoryResetCallbackReference(requestOptions));
```

Comment:
do we need to also set the context supplier here: 
```
           queryOptionsAccessor.setDiagnosticsContextSupplier(
                options,
                () -> pagedFluxOptions.getDiagnosticsContextSnapshot());
```

Reply:
I think addressed with the FeedOperationState refactoring